shared/hooks/useDateFilter.ts

```diff
@@ -0,0 +1,94 @@
+import { DateTime } from 'luxon'
+import { useState, useEffect, Dispatch, SetStateAction } from 'react'
+
+import { EventModel } from 'shared/api'
+
+type UseDateFilterReturn = {
+  fromDate: string
+  setFromDate: Dispatch<SetStateAction<string>>
+  toDate: string
+  setToDate: Dispatch<SetStateAction<string>>
+  filteredEvents: EventModel[]
+  fromDateError: string | null
+  toDateError: string | null
+}
+
+const useDateFilter = (events: EventModel[] | null, translation: (key: string) => string): UseDateFilterReturn => {
```

Comment:
This whole function should get a lot simpler too if you use `DateTime` and its methods.